### ISSUE TITLE
Multisplat16/Lite Shader normal calculation wrong + Fix

### ISSUE THREAD

**Issue by tinmanjuggernaut** [Zylann/godot_heightmap_plugin#308]
The lighting is wrong on the terrain due to an incorrect normal calculation. This problem exists in multisplat16 and multisplat16lite. I didn't see it in the others.

https://www.youtube.com/watch?v=Sv3pxzI0OcE

Note in the video I didn't see that the antitile was negating Y. What is shown in the video has issues at other angles. I played with a few more things and this is the final solution:

```
diff --git a/project/addons/zylann.hterrain/shaders/multisplat16.shader b/project/addons/zylann.hterrain/shaders/multisplat16.shader
index 77481268..9b44a1d0 100644
--- a/project/addons/zylann.hterrain/shaders/multisplat16.shader
+++ b/project/addons/zylann.hterrain/shaders/multisplat16.shader
@@ -262,8 +263,7 @@ vec4 texture_array_antitile(sampler2DArray albedo_tex, sampler2DArray normal_tex
        // Normals have to be rotated too since we are rotating the texture...
        // TODO Probably not the most efficient but understandable for now
        vec3 n = unpack_normal(nrm1);
-       // Had to negate the Y axis for some reason. I never remember the myriad of conventions around
-       n.xz = rotate(n.xz, cosa, -sina);
+       n.xz = -rotate(n.xz, cosa, sina);
        nrm1 = pack_normal(n, nrm1.a);

        // Periodically alternate between the two versions using a warped checker pattern
@@ -339,7 +340,7 @@ void fragment() {
        }

        vec3 terrain_normal_world =
-               u_terrain_normal_basis * (unpack_normal(texture(u_terrain_normalmap, UV)) * vec3(1,1,-1));
+               u_terrain_normal_basis * unpack_normal(texture(u_terrain_normalmap, UV));
        terrain_normal_world = normalize(terrain_normal_world);
        vec3 normal = terrain_normal_world;

```

**Comment by Zylann** [comment]
Man, 17min video just for that small error xD
I remember this existed in other shaders for a while, then was deleted. Maybe I forgot this one?

**Comment by Zylann** [comment]
After only removing the Z negation in the terrain normalmap, and giving proper normal maps to textures, this is what I get without antitile:
![image](https://user-images.githubusercontent.com/1311555/153721829-332d263d-e05e-4485-a102-285bf305d5d3.png)
And with antitile:
![image](https://user-images.githubusercontent.com/1311555/153721848-c098de5c-6cd0-47b6-8d82-229808effe3a.png)

**Comment by Zylann** [comment]
Should be fixed in f5c67242bd8e45d10b8e3607463134ca3d82e17d
